plugins/wallabag.koplugin/main.lua

```diff
@@ -0,0 +1,695 @@
+    -- workaround for dateparser, only once
+    -- the parser is in newsdownloader.koplugin, check if it is available
+    if not self.is_dateparser_checked then
+        local res
+        res, self.dateparser = pcall(require, "lib.dateparser")
+        if res then self.is_dateparser_available = true end
+        self.is_dateparser_checked = true
+    end
+end
+
+function Wallabag:addToMainMenu(menu_items)
+    menu_items.wallabag = {
+        text = _("Wallabag"),
+        sub_item_table = {
+            {
+                text = _("Retrieve new articles from server"),
+                callback = function()
+                    if not NetworkMgr:isOnline() then
+                        NetworkMgr:promptWifiOn()
+                        return
+                    end
+                    self:synchronise()
+                    self:refreshCurrentDirIfNeeded()
+                end,
+            },
+            {
+                text = _("Delete finished articles remotely"),
+                callback = function()
+                    if not NetworkMgr:isOnline() then
+                        NetworkMgr:promptWifiOn()
+                        return
+                    end
+                    local num_deleted = self:processLocalFiles( "manual" )
+                    UIManager:show(InfoMessage:new{
+                        text = T(_('Articles processed.\nDeleted: %1'), num_deleted)
+                    })
+                    self:refreshCurrentDirIfNeeded()
+                end,
+                enabled_func = function()
+                    return self.is_delete_finished or self.is_delete_read
+                end,
+            },
+            {
+                text = _("Go to download folder"),
+                callback = function()
+                    if FileManager.instance then
+                        FileManager.instance:reinit( self.directory )
+                    else
+                        FileManager:showFiles( self.directory )
+                    end
+                end,
+            },
+            {
+                text = "Settings",
+                callback_func = function()
+                    return nil
+                end,
+                sub_item_table = {
+                    {
+                        text = _("Configure Wallabag server"),
+                        keep_menu_open = true,
+                        callback = function()
+                            self:editServerSettings()
+                        end,
+                    },
+                    {
+                        text_func = function()
+                            local path
+                            if not self.directory or self.directory == "" then
+                                path = "Not set"
+                            else
+                                path = filemanagerutil.abbreviate(self.directory)
+                            end
+                            return _("Set download directory") .. " (" .. path .. ")"
```

Comment:
Even for stuff like this it's generally preferable to use T(). The thing is, you don't really know if in some localization might like to make it `(path) Set download directory` or `Download directory (path) set` or many other possibilities. :-)